### ISSUE TITLE
Future status on  implementing barcode readers support

### ISSUE THREAD

**Issue by IMiGS** [openboxes/openboxes#240]
Is there any near-future plans on implementing barcode reader support, for exampling for syncing system registered stock status  with the barcode reader scans (adding or subtracking stock) or getting a stock count and name by scanning registered items barcode?

**Comment by jmiranda** [comment]
Yeah, I've thought about adding the barcode support for years, but only recently had a user request it and provide concrete requirements (#205) for the feature. I haven't had any time to work on it, but I'm hoping to get to it in 2018.

Can you describe how it would be used? What are you barcoding? Lot numbers? Product code? Manufacturer code? Bin Location? 

Here's a scenario that I've considered but don't know if this is what people are thinking:

1. User logs into the system
1. System redirects to dashboard
1. User scans barcode (could be shipment number, product code, manufacturer code, etc)  
1. System locates object associated with barcode (in this case it's a lot number)
1. System redirects to stock card page for the product associated with the lot number
1. System prompts user for an action (if necessary)
1. User chooses to adjust stock
1. System pops up a modal dialog to allow user to adjust stock

The redirect to the stock card page might not be necessary. We could just keep you on the current page and pop up a modal dialog for the most likely action (i.e. Adjust Stock) or prompt the user for an action.

Eventually I'd also like to build an Android or hybrid app (#169 Road Map Ideas > Build prototype mobile app with ability to scan barcodes (using Ionic, phonegap, etc) that allows users to do most of this from their phone.

**Comment by IMiGS** [comment]
We have very simple idea using barcode themselfs, but it would be also great if it had also some kind a editor or option to import created barcode sticker templates, for example we would want to have a sticker like:
![image](https://user-images.githubusercontent.com/7812859/34477051-1550ccf0-ef9f-11e7-8fea-1e0e014b7eed.png)

On process itself:
I think it would need to have a simpler task selector not so much a GUI more like a simple text and table because it would give not only faster load time but also would work on different kind of devices.
Most of the time barcode reader will be used to change stock status, maybe there should have and option like what you want to do:

*1.Location selection- selects location
*2.Inspect item - gives info on object and stock
*3.Add item to storage - adds automatically to stock +1 item by barcode id
*4.Add item and edit it - shows item and allows to edit stock and item info itself
*5.Add new product to storage - creates a new item in the system
*6.Remove item - gives option to remove item from the stock
(if used on real barcode readers then 1-6 would be active links, for example, if I click on barcode scanner number 1 then it activated link nr.1 - Location selection)

But of course your proposed idea is also great and has all the necessary things, especially  it would work if it had Android/mobile phone support because phones has larger screens to view webpages and GUI information.
